flink/v1.15/flink/src/main/java/com/netease/arctic/flink/read/source/MORDataIterator.java

```diff
@@ -0,0 +1,56 @@
+package com.netease.arctic.flink.read.source;
+
+import org.apache.iceberg.io.CloseableIterator;
+
+import java.io.IOException;
+import java.util.Collections;
+
+/**
+ * Flink data iterator that reads {@link com.netease.arctic.scan.KeyedTableScanTask} into a {@link CloseableIterator}
+ *
+ * @param <T> is the output data type returned by this iterator.
+ */
+public class MORDataIterator<T> extends DataIterator<T> {
+
+  private CloseableIterator<T> iterator;
+
+  public MORDataIterator(CloseableIterator<T> iterator) {
+    super(null, Collections.emptyList(), null, null);
+    this.iterator = iterator;
```

Comment:
Should we consider consistency here and have the iterator created by this class instead of being passed in, as with DataIterator?